### ISSUE TITLE
Customizable Rollups: Skew Mode Single Result Operation Rollups are Cleared in Batch Job

### ISSUE THREAD

**Issue by jesslopez-sf** [SalesforceFoundation/Cumulus#4358]
This is affecting First Gift Amount, Last Gift Amount, Smallest / Largest , etc. When clicking the **_Recalculate Rollups_** button on an affected Contact/Account, it calculates these values correctly. However, when the batch job runs at night,  it clears them out. This is only occurring for the skew mode versions of the rollup jobs.

We are investigating a Case where this started occurring when NPSP 3.156 was pushed to the org. This release included optimizations and heap size reductions to how single result operation rollups are calculated/stored (#4232 ). 

### Steps to Repeat:

1. Enable Customizable Rollups in NPSP Settings → Donations → Customizable Rollups
2. Lower the skew mode batch sizes in NPSP Settings→ Bulk Data Processes → Batch Process Settings
    1. Set Skew Mode Dispatcher Batch Size to 3
    2. Set Account Skew mode Batch Size to 10
    3. Set Skew Mode Threshold to 50
3. Set Account History Tracking
    1. Go to Setup → Object Manager → Account → Fields and Relationships
    2. Click on Set History Tracking button.
    3. Check off Enable Account History and select the following fields: 
    4. First Gift Date, Last Gift Date, Smallest Gift, Largest Gift, Total Gifts, Total Number of Gifts (including last 2 to show that they don’t change)
    5. Save.
4. Create and save report of type “Account History”
5. Delete any existing test data in org
6. Insert 10 Contacts
```
List<Contact> contactList = new List<Contact>();

for(Integer x = 0; x < 10; x++)
{
    contactList.add(new Contact(
        FirstName = 'Test',
        LastName = 'Contact' + x
    ));
}

insert contactList;
```
7. Insert 100 Opportunities per Contact (Run 10 times to avoid timeouts, but make sure crlps job has finished between each run)
```
List<Account> acctList = [SELECT Id FROM Account WHERE npo02__NumberOfClosedOpps__c = 0 LIMIT 1];
List<Opportunity> opps = new List<Opportunity>();

for(Integer x = 0; x < 100; x++)
{
    opps.add(new Opportunity(
        Name = 'test ' + x,
        CloseDate = System.today().addDays(-x),
        StageName = 'Closed Won',
        AccountId = acctList[0].Id,
        Amount = x + 1
    ));
}

insert opps;
```

### Tests

**Test 1: The Control**
1. Confirm that all Accounts have values in First Gift Date, Last Gift Date, etc.
2. Go to NPSP Settings → Bulk Data Processes → Rollup Donations Batch and click the Run Batch button.
![Screen Shot 2019-07-01 at 4 06 21 PM](https://user-images.githubusercontent.com/29731456/60464602-0d24a200-9c1d-11e9-8585-ec7786280914.png)

**Test 2: The Bug
 Insert Opportunities that should trigger an update to the First Gift field**

```
List<Account> accts = [SELECT Id from Account];
List<Opportunity> opps = new List<Opportunity>();


for(Account a:accts)
{
    opps.add(new Opportunity(
        Name = 'old opp',
        CloseDate = Date.newInstance(2000, 01,01),
        Stagename ='Closed Won',
        Amount = 100,
        AccountId = a.Id
    
    ));
}
insert opps;
```
1. In Account History report, confirm that the First Gift Date was updated on all Accounts.
2. Run the Rollup Donations batch again and now confirm that the single value rollup operation-based values have been cleared.
![Screen Shot 2019-07-01 at 4 06 28 PM](https://user-images.githubusercontent.com/29731456/60464643-262d5300-9c1d-11e9-8a23-9b9cca2acc33.png)

3. Go to a single Account record and click the Recalculate Rollups button. Confirm that the values are populated again (after refreshing a few times, because of Lightning)

**Comment by jesslopez-sf** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001C6wBQAS" target="_blank">W-034147</a>

**Comment by mrbelvedere** [comment]
Included in beta release 3.158 (Beta 7)

**Comment by mrbelvedere** [comment]
Included in production release 3.158
